retrieve-logs/retrieve.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ -z "${START_DATE}" ]; then
+    echo "Please ensure you use the \"START_DATE\" variable to specify a start date to begin searching from."
+    echo "Only a date should be provided, in the format 'yyyy-mm-dd'."
+    echo "The script assumes logs should be searched from the start of this day, in a UTC timezone."
+    exit 1
+fi
+if [ -z "${NUM_OF_DAYS}" ]; then
+    echo "Number of days not specified. Will search for 1 day of logs."
+    export NUM_OF_DAYS=1
+fi
+
+set -u
+
+function be_in_directory {
```

Comment:
Doesn't need changing, just wondering why you did this as function when it's only used once?

Reply:
1. I'm lazy. I have this as a snippet so I can copy/paste as necessary.
2. This feels like it fits the "self-documenting code" paradigm better. The alternative to me would be to do a 1-liner, but add a comment to explain what it's actually doing. (Because just reading shell scripts, I often have no clue... 😅 )